engine/script/src/script.cpp

```diff
@@ -258,13 +258,76 @@ namespace dmScript
         Unref(L, LUA_REGISTRYINDEX, context->m_ContextTableRef);
     }
 
-    lua_State* GetLuaState(HContext context) {
+    lua_State* GetLuaState(HContext context)
+    {
         if (context != 0x0) {
             return context->m_LuaState;
         }
         return 0x0;
     }
 
+    // From ldblib.c (getthread)
+    static lua_State* GetLuaThread(lua_State *L, int *arg)
+    {
+        if (lua_isthread(L, 1)) {
+            *arg = 1;
+            return lua_tothread(L, 1);
+        }
+        else {
+            *arg = 0;
+            return L;
+        }
+    }
+
+    // From https://zeux.io/2010/11/07/lua-callstack-with-c-debugger/
+    // and also
+    // https://github.com/defold/defold/blob/dev/engine/lua/src/lua/ldblib.c#L321
+    void GetLuaTraceback(lua_State* L, const char* infostring, void (*cbk)(lua_State* L, lua_Debug* entry, void* ctx), void* ctx)
```

Comment:
The new function.
Compare with https://github.com/defold/defold/blob/dev/engine/lua/src/lua/ldblib.c#L321